modules/kafka/src/main/scala/higherkindness/mu/kafka/package.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package higherkindness.mu
+
+import cats.effect.{ConcurrentEffect, ContextShift, Sync, Timer}
+import fs2.{Pipe, Stream}
+import fs2.kafka.{AutoOffsetReset, ProducerRecords, ProducerResult, ProducerSettings}
+import higherkindness.mu.format.{Decoder, Encoder}
+import higherkindness.mu.kafka.config.KafkaBrokers
+
+package object kafka {
+  type ByteArrayProducerResult  = ProducerResult[String, Array[Byte], Unit]
+  type ByteArrayProducerRecords = ProducerRecords[String, Array[Byte], Unit]
+  type PublishToKafka[F[_]] =
+    Pipe[F, ByteArrayProducerRecords, ByteArrayProducerResult]
+
+  object consumerSettings {
+    def atLeastOnceFromEarliest[F[_]: Sync](
+        groupId: String,
+        brokers: KafkaBrokers
+    ): fs2.kafka.ConsumerSettings[F, String, Array[Byte]] =
+      fs2.kafka
+        .ConsumerSettings[F, String, Array[Byte]]
+        .withGroupId(groupId)
+        .withBootstrapServers(brokers.urls)
+        .withAutoOffsetReset(AutoOffsetReset.Earliest)
+        .withEnableAutoCommit(true)
+  }
+
+  def consumer[F[_], A](
+      topic: String,
+      groupId: String,
+      messageProcessingPipe: Pipe[F, A, A]
+  )(
+      implicit contextShift: ContextShift[F],
+      concurrentEffect: ConcurrentEffect[F],
+      timer: Timer[F],
+      decoder: Decoder[A],
+      brokers: KafkaBrokers
+  ): F[Unit] =
+    ConsumerStream(topic, consumerSettings.atLeastOnceFromEarliest(groupId, brokers))
+      .through(messageProcessingPipe)
+      .compile
```

Comment:
Shouldn't we keep the result of `consumer` as `Stream[F, A]`. In this way, the user could consume a message, process it and then produce this new message on a new topic 🤔